File: 1636-number-of-substrings-with-only-1s/number-of-substrings-with-only-1s.cpp
class Solution {
    const int MOD = 1e9 + 7;
    using ll = long long;
public:
    int numSub(string s) {
        int n = s.size();

        int i = 0, j = 0;
        ll result = 0;
        while (i < n) {
            j++;
            if (s[i] == '0') {
                i++;
            } else {
                while (j < n && s[i] == s[j]) {
                    j++;
                }

                result += ((j - i) * 1LL * (j - i + 1)) / 2;
                i = j;
            }
        }

        return result % MOD;
    }
};

File: 1636-number-of-substrings-with-only-1s/number-of-substrings-with-only-1s.java
class Solution {
    final int MOD = (int)1e9 + 7;
    public int numSub(String s) {
        int n = s.length();

        int i = 0, j = 0;
        long result = 0;
        while (i < n) {
            j++;
            if (s.charAt(i) == '0') {
                i++;
            } else {
                while (j < n && s.charAt(i) == s.charAt(j)) {
                    j++;
                }

                result += ((j - i) * 1L * (j - i + 1)) / 2;
                i = j;
            }
        }

        return (int)(result % MOD);
    }
}